### PR TITLE
Feature/juan/remote mediator

### PR DESCRIPTION
**Pull request by juancoob**
**ISSUE**
#60 

**SUMMARY**
Added the remote mediator to paginate with the database and the server

### CODE REVIEW

**Review by jujoru — APPROVED** [review]

File: app/src/main/java/com/architectcoders/arquitectomarvel/data/database/RoomDataSource.kt [hunk 1]
```diff
@@ -10,19 +11,44 @@ class RoomDataSource(db: MarvelDatabase) : LocalDataSource {
 
     private val characterDao = db.marvelDao
 
-    override suspend fun getLocalCharacters(): List<CharacterResult> =
+    override suspend fun getLocalCharacters(): List<CharacterResult>? =
         withContext(Dispatchers.IO) {
```
Comment (joseantoniocampillo) [inline app/src/main/java/com/architectcoders/arquitectomarvel/data/database/RoomDataSource.kt]: I think that wrap with withContext(Dispatcher.IO) is a boilerplate unnecessary.
We can see in google repos like [this](https://github.com/android/sunflower)

**Review by joseantoniocampillo — APPROVED** [review]
Good work.
I see only we don't need wrap call to room changing dispatchers.

File: data/src/main/java/com/architectcoders/data/repository/MarvelRepository.kt [hunk 2]
```diff
@@ -11,32 +11,44 @@ class MarvelRepository(
...
-    }
+    suspend fun getRemoteCharacters(offset: Int): Characters =
+        remoteDataSource.getRemoteCharacters(offset)
 
-    suspend fun getRemoteCharacterById(characterId: Int): Characters =
-        remoteDataSource.getRemoteCharacterById(characterId)
+    suspend fun getStoredCharactersCount(): Int =
+        localDataSource.getStoredCharactersCount()
+
+    suspend fun getRemoteCharacterById(characterId: Int): CharacterResult =
```
Comment (joseantoniocampillo) [inline data/src/main/java/com/architectcoders/data/repository/MarvelRepository.kt]: Change name of function by getLocalCharacterById

**Review by joseantoniocampillo — CHANGES_REQUESTED** [review]
Please, look the comments

File: data/src/main/java/com/architectcoders/data/repository/MarvelRepository.kt (hunk 2 above)
Comment (juancoob) [reply data/src/main/java/com/architectcoders/data/repository/MarvelRepository.kt]: Sure :)

File: app/src/main/java/com/architectcoders/arquitectomarvel/data/database/RoomDataSource.kt (hunk 1 above)
Comment (juancoob) [reply app/src/main/java/com/architectcoders/arquitectomarvel/data/database/RoomDataSource.kt]: You are right @joseantoniocampillo ! Check [PlantRepository.kt](https://github.com/android/sunflower/blob/main/app/src/main/java/com/google/samples/apps/sunflower/data/PlantRepository.kt) it says: _Room supports Coroutines and moves the query execution off of the main thread_. Thanks!

**Review by joseantoniocampillo — APPROVED** [review]
ok

**Pull request merged**